Sources/KlaviyoSwift/APIRequestErrorHandling.swift

```diff
@@ -12,12 +12,48 @@ struct ErrorHandlingConstants {
     static let maxBackoff = 60 * 3 // 3 minutes
 }
 
+enum InvalidField: Equatable {
+    case email
+    case phone
+
+    /// gets the invalid field based on the source.pointer from klaviyo API.
+    /// this assumes that source.pointer will not change
+    /// Client APIs to have better error codes in the future at which point we should use that instead of source.pointer
+    /// - Parameter sourcePointer: pointers to the source of the error
+    /// - Returns: the field that is invalid else `nil`
+    static func getInvalidField(sourcePointer: String) -> InvalidField? {
+        if sourcePointer == "/data/attributes/phone_number" {
+            return .phone
+        }
+        if sourcePointer == "/data/attributes/email" {
+            return .email
+        }
+
+        return nil
+    }
+}
+
 private func getDelaySeconds(for count: Int) -> Int {
     let delay = Int(pow(2.0, Double(count)))
     let jitter = environment.randomInt()
     return min(delay + jitter, ErrorHandlingConstants.maxBackoff)
 }
 
+private func parseError(_ data: Data) -> [InvalidField]? {
+    var invalidFields: [InvalidField]?
```

Comment:
Not clear from the diff but is this happening on a background thread?

Reply:
It's the same queue that `handleRequestError` is called which seems to have [user-initiated QoS](https://developer.apple.com/documentation/dispatch/dispatchqos/1780759-userinitiated). 
![image](https://github.com/klaviyo/klaviyo-swift-sdk/assets/118314354/ee75dbec-cb38-49f6-8935-da9a8827e0ea)